# Blatt 2 

## Nr. 1 

Da in allen Aufgabenteilen mit zwei Würfeln gewürfelt wurde. Somit ergibt sich dann als Wahrscheinlichkeit für einen einzigen beliebigen Wurf mit einem Würfel $P(\text{ein Würfel, beliebiges Ergebnis})= \frac{1}{6}$ und für einen Wurf mit zwei Würfeln ergibt sich dann $P(\text{zwei Würfel, beliebiges Ergebnis})= \frac{1}{36}$. Diese Annahme wurde in den folgenden Teilaufgaben benutzt. 

#### a) $P(W_\text{rot} + W_\text{blau} = 9) =\\P(W_\text{rot}= 3 ,W_\text{blau} = 6) + P(W_\text{rot}= 4 ,W_\text{blau} = 5) + P(W_\text{rot}= 5 ,W_\text{blau} = 4) + P(W_\text{rot}= 6 ,W_\text{blau} = 3)\\= \frac{4}{36} = \frac{1}{9}$

#### b) $P(W_\text{rot} + W_\text{blau} \geq 9) = \\P(W_\text{rot}= 3 ,W_\text{blau} = 6) + P(W_\text{rot}= 4 ,W_\text{blau} = 6) + P(W_\text{rot}= 5 ,W_\text{blau} = 6) + P(W_\text{rot}= 6 ,W_\text{blau} = 6) + \\P(W_\text{rot}= 4 ,W_\text{blau} = 5) + P(W_\text{rot}= 5 ,W_\text{blau} = 5) + P(W_\text{rot}= 6 ,W_\text{blau} = 5) + \\P(W_\text{rot}= 5 ,W_\text{blau} = 4) + P(W_\text{rot}= 6 ,W_\text{blau} = 4) + \\P(W_\text{rot}= 6 ,W_\text{blau} = 3) \\= \frac{10}{36} = \frac{5}{18}$

#### c) $P(W_\text{rot}=4, W_\text{blau}=5 \lor W_\text{rot}=5, W_\text{blau}=4) =\\ P(W_\text{rot}= 4 ,W_\text{blau} = 5) + P(W_\text{rot}= 5 ,W_\text{blau} = 4)\\= \frac{2}{36} = \frac{1}{18}$

#### d) $P(W_\text{rot}=4, W_\text{blau}=5) = \frac{1}{36}$

In den folgenden Aufgaben steht schon fest, dass der rote Würfel eine $4$ zeigt, insofern wird dann quasi nur noch ein einzelner Wurf des blauen Würfels betrachtet.

#### e) $P(W_\text{rot} + W_\text{blau} = 9) \\= P(W_\text{blau} = 5)\\= \frac{1}{6} $

#### f) $P(W_\text{rot} + W_\text{blau} \geq 9) \\= 
P(W_\text{blau} = 5) + P(W_\text{blau} = 6) \\= \frac{2}{6} = \frac{1}{3}$

#### g) $P(W_\text{rot}= 4, W_\text{blau} = 5) \\= P(W_\text{blau} = 5) \\= \frac{1}{6}$

# Aufgabe 2

## Bestimmung der Normalisierungskonstante

Gegeben war 
\begin{align}
    f(v)=N \cdot exp\left(\frac{m v^2}{2k_\text{B}T}\right)4 \pi v^2 \,.
\end{align}
Mit der Bedingung
\begin{align}
    \int_{0}^{\infty}f(v)\,\text{d}v=1
\end{align}
folgt dann:
\begin{align}
    \frac{1}{4\pi N}&=\int_0^{\infty} \exp\left(\frac{m v^2}{2k_\text{B}T}\right) v^2 \, \text{d} v \\
    \frac{1}{4\pi N}&=-2k_\text{B}T \frac{\text{d}}{\text{d}m} \int_{0}^{\infty}\exp\left(\frac{m v^2}{2k_\text{B}T}\right) \,\text{d}v\\
    \frac{1}{4\pi N}&=-2k_\text{B}T \frac{\text{d}}{\text{d}m} \sqrt{\frac{2k_\text{B}T}{m}} \int_{0}^{\infty} \exp(-u^2)\, \text{d} u\\
    \frac{1}{4\pi N}&=-2k_\text{B}T \frac{\text{d}}{\text{d}m} \sqrt{\frac{2k_\text{B}T}{m}} \frac{\sqrt{\pi}}{2}\\
    \frac{1}{N}&=8 \pi k_\text{B}T \sqrt{\frac{\pi k_\text{B} t}{2}} \frac{1}{2m^{3/2}}=\left(\frac{2\pi k_\text{B} T}{m}\right)^{3/2} \\
    N&=\left(\frac{m}{2 \pi k_\text{B} T}\right)^{3/2}
\end{align}

## Aufgabe 2a)
Die wahrscheinlichste Geschwindigkeit $v_m$ entspricht dem Maximum der Wahrscheinlichkeitsdichtefunktion der Geschwindigkeit. Die erste Ableitung muss also null sein:

\begin{align}
    \frac{\text{d}f(v)}{\text{d}v}&=4 \pi N v \exp\left(\frac{m v^2}{2k_\text{B}T}\right)\left(2-\frac{mv^2}{k_\text{B}T}\right) \\
    \rightarrow 2&=\frac{mv^2}{k_\text{B}T} \\
    \rightarrow v^2&= \frac{2k_\text{B}T}{m} \\
    \rightarrow v_m&=\sqrt{\frac{2k_\text{B}T}{m}} 
\end{align}

Damit kann man $f(v)$ umschreiben zu:
\begin{equation}
    f(v)=\frac{4}{\sqrt{\pi}}\frac{1}{v_m^3} v^2 \exp\left(-\left(\frac{v}{v_m}\right)^2\right)
\end{equation}

## Aufgabe 2b)
Gesucht ist der Mittelwert $\langle v \rangle$ der Geschwindigkeit.

\begin{align}
    \langle v \rangle &= \int_0^{\infty} v\, f(v)\, \text{d}v \\
    &=\frac{4}{\sqrt{\pi}\,v_m^3} \int_0^{\infty} v^3 \exp\left(-\left(\frac{v}{v_m}\right)^2\right) \\
    &= \frac{4}{\sqrt{\pi}\,v_m^3} \frac{v_m^4}{2} \\
    &=\frac{2 v_m}{\sqrt{\pi}}
\end{align}

## Aufgabe 2c)
Gesucht ist der Median $v_{0,5}$ der Geschwindigkeitsverteilung. Dieser kann berechnet werden durch 
\begin{equation}
    \frac{1}{2}=\int_0^{v_{0,5}}f(v)\, \text{d}v=\frac{4}{\sqrt{\pi}v_m^3}\int_0^{v_{0,5}} v^2 \exp\left(-\left(\frac{v}{v_m}\right)^2\right)\, \text{d}v
\end{equation}
Umformungen ergeben
\begin{equation}
    1=\frac{8}{\sqrt{\pi}} \int_0^{\frac{v_{0,5}}{v_m}} u^2 exp(-u^2)\, \text{d}u
\end{equation}
Das muss jetzt numerisch gelöst werden.

In [1]:
import numpy as np 
from scipy.integrate import quad
from uncertainties import ufloat

#Definition der Funktion, Koeffizient auf andere Seite gebracht
def func(u):
    return u**2 *np.exp(-u**2)

v = np.linspace(0, 10, 10**5) # Geschwindigkeiten v werden erzeugt

solution =  np.sqrt(np.pi)/8 #linke Seite 
g2 = np.empty(1) #leeres Array für for-loop erzeugen (nicht schön gelöst. aber funktioniert.)

# mit quad Funktion aus scipy.integrate wird die Funktion numerisch integriert und es wird ein Array g mit 
# dem Integral über die Werte v befüllt
for i in range(len(v)): 
    g = np.append(g2,quad(func, 0, v[i])[0])
    g2 = g 
g = np.delete(g, 0) #Die erste Komponente des Hilfsarray wird gelöscht

v_bigger_than_median = v[g > solution] # alle v die größer als der Median sind
v_smaller_than_median = v[g < solution] # alle v die kleiner als der Median sind 
v_median_array = np.array([v_bigger_than_median[0], v_smaller_than_median[-1]]) #Mittelwert zwischen den beiden Werten wird gebildet
v_median = ufloat(v_median_array.mean(), v_median_array.std())
print(f"Die numerisch ermittelte Grenze beträgt: {v_median}")

Die numerisch ermittelte Grenze beträgt: 1.08766+/-0.00005


Dieser Wert muss aufgrund der Grenzen Änderung noch mit $v_m$ multipliziert werden. Es ergibt sich also als Median ein ungefährer Wert von 

\begin{align}
v_{0.5}= 1.08766 \cdot v_m .
\end{align}

## Aufgabe 2d) 

Gesucht wird die volle Breite auf halber Höhe der Verteilung ($v_\text{FWHM}$). 

Dabei gilt 

\begin{align}
f(v) &= \frac{f(v_m)}{2} \\
\frac{1}{2} &= \frac{f(v)}{f(v_m)}  \\
\frac{1}{2} &= \frac{v}{v_m}^2 \exp\left(- \left( \left(\frac{v}{v_m}\right)^2 -1 \right) \right) 
\end{align}

Substituiert man $\frac{v}{v_m}^2$ zu $u$ ergibt sich die Form 

\begin{align}
u \cdot exp(-u) &= \frac{1}{2 e}.
\end{align}

Die Lösung dieses Formalismus lässt sich mittels der Lambertschen W-Funktion lösen. 
Diese ist definiert als 

\begin{align} 
 z = W(z)\cdot exp (W (z))
\end{align} 

und gibt uns somit eine Lösung zu unserem Problem. Es ergibt sich dann, dass 

\begin{align}
u = -W_k \left( -\frac{1}{2e} \right) \text{ mit } k \in \mathbb{Z}
\end{align} 
ist. Somit ist dann natürlich 
\begin{align}
v = \sqrt{-W_k \left( -\frac{1}{2e} \right) }\, v_m .
\end{align}

Dies sind also die Stellen an denen die Funktion auf der Hälfte ihrer Werte landet. 
Es ergeben sich zwei analytische Lösungen für u. 

\begin{align}
u_1 &= 0.231961 \\
u_2 &= 2.67835
\end{align}

Mit der Differenz dieser beiden Werte kommen wir dann auf unsere volle Breite auf halber Höhe. 

\begin{align}
v = \left(\sqrt{ u_2} - \sqrt{u_1}\right) \cdot v_m = 1.155 \cdot v_m 
\end{align}


## Aufgabe 2e)  

Gesucht ist die Standardabweichung der Geschwindigkeit $\sigma_\text{v}$. 
Dafür benötigen wir den Mittelwert des Geschwindigkeit Quadrats $\langle v^{2} \rangle$ der Geschwindigkeit.

\begin{align}
    \langle v \rangle &= \int_0^{\infty} v\, f(v)\, \text{d}v \\
    &=\frac{4}{\sqrt{\pi}\,v_m^3} \int_0^{\infty} v^4 \exp\left(-\left(\frac{v}{v_m}\right)^2\right) \\
    &= \frac{4}{\sqrt{\pi}\,v_m^3} \frac{3 \sqrt{\pi} v_m^5}{8} \\
    &=\frac{3 v_m^{2}}{2}
\end{align}

Die Standardabweichung ist als 

\begin{align}
        \sigma_\text{v} &= \sqrt{ \langle v^{2} \rangle - \langle v \rangle^{2} }
\end{align}

definiert. 

Somit ergibt sich dann 

\begin{align}
        \sigma_\text{v} &= \sqrt{ \langle v^{2} \rangle - \langle v \rangle^{2} } \\
        &= \sqrt{ \frac{3 v_m^{2}}{2} -  \frac{4 v_m^{2}}{\pi} } \\
        &= \left(\sqrt{ \frac{3}{2} - \frac{4}{\pi} }\right) \, v_m
\end{align} 
    